adr/ADR-43.md

```diff
@@ -22,19 +22,21 @@ Related issues [#3268](https://github.com/nats-io/nats-server/issues/3268)
 
 ## Per-Message TTL
 
-We will allow a message to supply a TTL using a header called `Nats-TTL` followed by the duration as seconds.
+We will allow a message to supply a TTL using a header called `Nats-TTL` followed by the duration as seconds or as a Go duration string like `1h`.
 
 The duration will be used by the server to calculate the deadline for removing the message based on its Stream 
 timestamp and the stated duration.
 
 The TTL may not exceed the Stream MaxAge. The shortest allowed TTL would be 1 second. When no specific TTL is given
```

Comment:
Currently we don't implement limiting the TTL to the `MaxAge`, which I think makes sense for two reasons:
1. The `MaxAge` could be edited later, but by that time we have TTL'd messages already that could exceed the new `MaxAge`
2. We already support a TTL `never`

Reply:
removed that line